enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,335 @@
+
+### Test Plan
+
+The new config API will require updates of the existing Insights operator testsuite. The testsuite will be extended to cover the new API and respective attributes. 
+Following behavior will need to be tested:
+- `support` secret config takes precedence over the new config API
+- forced gathering works as expected
+- disabling some gatherers works as expected and diasbling all will disable the gathering completely
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API behind the `TechPreviewNoUpgrade` feature gate. We would like to
+keep the original way of configuring the Insights operator (that is via `support` secret), because it is problematic to move every existing configuration option to the new configuration API and the existing `support` secret serves basically as a configuration API. 
+
+#### Dev Preview -> Tech Preview
+
+The original way of configuring the Insights is still available. If there is a conflict, in other words, if there is some configuration option defined in the `support` secret as well as in the new configuration API then the value from the `support` secret takes precedence. 
+
+#### Tech Preview -> GA
+
+The `TechPreviewNoUpgrade` feature gate requirement is removed. The behavior defined above in the [Dev Preview -> Tech Preview](#dev-preview---tech-preview) section is still true. 
+
+#### Removing a deprecated feature
+
+The existing configuration via `support` secret should be still available for backward compatibility reasons. The reason of the `support` secret existence is the option of using basic authentication for the Ingress service, which Insights operator uses as the upload endpoint. 
+
+### Upgrade / Downgrade Strategy
+
+No update of the configuration is required. The existing configuration in the `support` secret continues to work. When there is an existing `insights.config.openshift.io` kind in a cluster after upgrade, then the user can optionally use the new configuration API. 
+There can be a conflict when a configuration option is defined in the `support` secret as well as in the configuration API. The configuration in the `support` secret takes precedence in this case. 
+
+### Version Skew Strategy
+
+### Operational Aspects of API Extensions
+
+This API is provided exclusively for the Insights operator and there will be no other component depending on this new API. Current configobserver
+in the Insights operator runs every 5 minutes so if we preserve the same behavior then there will be a request to the API every 5 minutes. Additionally the field
+for the forced gathering (`ForceGatherReason` - see [Proposal](#proposal)) will require an informer.
```

Comment:
I wouldn't expect API call every 5 mins, you'll open a watch (using informer) which will notify you of all the changes.

Reply:
Updated